.devcontainer/devcontainer.json

```diff
@@ -0,0 +1,29 @@
+// For format details, see https://aka.ms/devcontainer.json. For config options, see the README at:
+// https://github.com/microsoft/vscode-dev-containers/tree/v0.183.0/containers/javascript-node
+{
+	"name": "Node.js",
+	"build": {
+		"dockerfile": "Dockerfile",
+		// Update 'VARIANT' to pick a Node version: 12, 14, 16
+		"args": { "VARIANT": "16" }
+	},
+
+	// Set *default* container specific settings.json values on container create.
+	"settings": {},
+
+	// Add the IDs of extensions you want installed when the container is created.
+	"extensions": [
+		"dbaeumer.vscode-eslint",
+    "editorconfig.editorconfig",
+    "davidanson.vscode-markdownlint"
+	],
+
+	// Use 'forwardPorts' to make a list of ports inside the container available locally.
+	"forwardPorts": [4000, 35729],
+
+	// Use 'postCreateCommand' to run commands after the container is created.
+	"postCreateCommand": "npm install && sudo gem install bundler && cd docs && bundle install",
```

Comment:
probably even append `&& bundle exec jekyll serve --livereload`?
if so, needs mention in the `.devcontainer/README.md` that a manual jekyll start is not required

Reply:
unfortunately it's not that easy :(

* the postCreateCommand runs just once during the first time setup
* vscode overwrites the entrypoint of the container to have a running process during the setup
   see overrideCommand at https://code.visualstudio.com/docs/remote/devcontainerjson-reference 
* set overrideCommand to false and add bundle exec jekyll serve --livereload as ENTRYPOINT in the dockerfile doesn't work either as the postCreateCommand did not run yet.. :(